tasks/main.yml

```diff
@@ -0,0 +1,104 @@
+- name: install apt packages
+  apt:
+    name: "{{ item }}"
+    state: present
+  with_items: "{{ LOAD_BALANCER_APT_PACKAGES }}"
+
+- name: create haproxy configuration directories
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: "0700"
+  with_items:
+    - "{{ LOAD_BALANCER_CERTS_DIR }}"
+    - "{{ LOAD_BALANCER_CONF_DIR }}"
+    - "{{ LOAD_BALANCER_BACKENDS_DIR }}"
+    - "{{ LETSENCRYPT_ARCHIVE_DIR }}"
+
+- name: copy haproxy configuration main section
+  template:
+    src: haproxy.cfg
+    dest: "{{ LOAD_BALANCER_CONF_DIR }}/00-main"
+
+- name: create placeholder file for backend map
+  copy:
+    content: |
+      # This directory contains files mapping domain names to backend names.  Each
+      # line contains a domain name and a backend name separated by whitespace.
+    dest: "{{ LOAD_BALANCER_BACKENDS_DIR }}/00-main"
+
+# Haproxy refuses to start when there are no valid SSL certificates in the
+# /etc/haproxy/certs directory, so we copy a bogus cert there.  The snakeoil
+# certificate will never be used since we used the strict-sni option in the
+# haproxy configuration.
+- name: copy snakeoil certificate into haproxy config directory
+  shell: >
+    cat /etc/ssl/certs/ssl-cert-snakeoil.pem /etc/ssl/private/ssl-cert-snakeoil.key \
+        > {{ LOAD_BALANCER_CERTS_DIR }}/ssl-cert-snakeoil.pem
+
+- name: copy letsencrypt staging certificate for testing purposes
+  copy:
+    src: lets-encrypt-fake-cert.crt
+    dest: "{{ LETSENCRYPT_FAKE_CERT }}"
+  when: LETSENCRYPT_USE_STAGING
+
+- name: copy helper scipts
+  template:
+    src: "{{ item }}"
+    dest: "/usr/local/sbin/{{ item }}"
+    mode: "0755"
+  with_items:
+    - haproxy-config-watcher
+    - haproxy-reload
+    - manage_certs.py
+
+- name: copy config watcher systemd service file
+  template:
+    src: haproxy-config-watcher.service
+    dest: /lib/systemd/system/haproxy-config-watcher.service
+
+- name: enable and start config watcher
+  command: systemctl {{ item }} haproxy-config-watcher
+  with_items:
+    - enable
+    - restart
```

Comment:
Nit: Which Ansible version range is this playbook looking to support? 2.2+ has a systemd module: https://docs.ansible.com/ansible/systemd_module.html